lib/input.rb

```diff
@@ -1,19 +1,23 @@
-require_relative './input_validator'
+require_relative 'command_line_input'
+require_relative 'input_validator'
+require_relative 'prompt'
 
 class Input
-    attr_accessor :input_type
-    def initialize(input_type)
-        @input_type = input_type
-    end
+  attr_reader :input_method, :validator
+
+  def initialize(input_method = CommandLineInput.new, validator = InputValidator.new)
```

Comment:
Why are you passing these in here? As opposed to elsewhere, when the `Input` class gets created.

Reply:
I wanted to provide some defaults. As a default, Input will get token inputs from the command line.

Reply:
Why not create them inside of the initializer then, as opposed to as arguments to the initializer?

Reply:
As such: 
def initialize(input_method, validator)
  @input_method = CommandLineInput.new
  @validator = InputValidator.new
end
??